src/Estimators/InputSection.h

```diff
@@ -137,6 +152,17 @@ class InputSection
   }
 
 protected:
+  /** reads attributes for both the root node and parameter/child nodes
+   *  that aren't delegated.
+   *
+   *  Side effect only method updates values_.
+   *  \param[in]   cur              current xml node
+   *  \param[in]   consume_name     read the name attribute (this has complicated semantics in QMCPACK input)
+   *  \param[in]   element_name     qualifying identifier with respect to the InputSection root node for the atttributes.
+   
+   *  Ideally any child node of significant complexity would be delegated to another input section.
+   */
+  void readAttributes(xmlNodePtr cur, bool consume_name, const std::string& element_name);
```

Comment:
I have difficulty to understand `consume_name` which is a boolean.

Reply:
added more documentation for this. This boolean reflects the inconsistent semantics of "name" attributes in our input.  It's troublesome but I don't see a way around this as long as we allow this way of writing a parameter input.
```
<parameter name="actual_parameter_type" some_parameter_attribute="whatever">...</parameter>
```